cpp/include/rapidsmpf/shuffler/chunk.hpp

```diff
@@ -94,6 +99,17 @@ class Chunk {
      */
     [[nodiscard]] std::unique_ptr<std::vector<uint8_t>> to_metadata_message() const;
 
+    /**
+     * @brief Serializes this chunk into a given metadata message buffer.
+     *
+     * @param msg The metadata message as a serialized byte vector.
+     * @param offset The offset in the message buffer to start writing.
+     * @returns The number of bytes written to the message buffer.
+     */
+    [[nodiscard]] std::ptrdiff_t to_metadata_message(
+        std::vector<uint8_t>& msg, std::ptrdiff_t offset
```

Comment:
Here if we accepted a `cuda::std::span` (C++20 but backported to C++14) over the message we wouldn't need offset argument separately, I think.